chain-crypto/src/algorithms/vrf/mod.rs

```diff
@@ -84,3 +84,20 @@ impl VerifiableRandomFunction for RistrettoGroup2HashDh {
         vr.u.clone()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    /// `secret_from_binary` should fail if the provided byte array does not match the public key size
+    fn vtf_size_checks() {
+        for n in 0..u8::MAX as usize {
+            let secret_key = RistrettoGroup2HashDh::secret_from_binary(&vec![0; n]);
```

Comment:
as discussed, let's try to rewrite test so it uses arbitrary for size of input vector

Reply:
Done. I also changed the test slightly to ensure we don't have any false `SizeInvalid` instances.